client/src/pages/user-preferences/UserPreferences.js

```diff
@@ -0,0 +1,72 @@
+import Form from "react-bootstrap/Form";
+import Button from "react-bootstrap/Button";
+import { useForm } from "react-hook-form";
+import Data from "../../components/form-data/formdata";
+import "./UserPreferences.css";
+
+export default function UserPreferences() {
+	const {
+		register,
+		handleSubmit,
+		formState: { errors },
+	} = useForm();
+	const onSubmit = (data) => {
+	};
+	return (
+		<Form className="form" onSubmit={handleSubmit(onSubmit)}>
+			<Form.Group className="form-group2" controlId="service">
+				<Form.Label>Please select the service:</Form.Label>
+				<Form.Select
+					aria-label="Default select example"
+					className="w-50"
+					type="text"
+					name="service"
+					{...register("service", {
+						required: "Please select your service",
+					})}
+				>
+					<option value="">Select here...</option>
+					{Data.services.map((service) => {
+						return (
+							<option key={service} value={service}>
+								{service}
+							</option>
+						);
+					})}
+				</Form.Select>
+				<Form.Text className="error-message">
```

Comment:
There is some built-in functionality in Boostrap to achieve this.
Here is an example usage https://github.com/Miladepour/SyntaxSquad-Project-Kunji/pull/16/files#diff-d2e54251fe00b6c1aa216984652dbe359d580b1379364133310be76e0250289bR55

The logic to show it is contolled via the `isInvalid` from the `<Form.Control>`

Reply:
Solved.